src/NuGet.Services.AzureSearch/Db2AzureSearch/Db2AzureSearchCommand.cs

```diff
@@ -20,28 +19,22 @@ public class Db2AzureSearchCommand
         private readonly INewPackageRegistrationProducer _producer;
         private readonly IIndexActionBuilder _indexActionBuilder;
         private readonly ISearchServiceClientWrapper _serviceClient;
-        private readonly ISearchIndexClientWrapper _searchIndexClient;
-        private readonly ISearchIndexClientWrapper _hijackIndexClient;
-        private readonly IVersionListDataClient _versionListDataClient;
+        private readonly Func<IBatchPusher> _batchPusherFactory;
         private readonly IOptionsSnapshot<Db2AzureSearchConfiguration> _options;
         private readonly ILogger<Db2AzureSearchCommand> _logger;
 
         public Db2AzureSearchCommand(
             INewPackageRegistrationProducer producer,
             IIndexActionBuilder indexActionBuilder,
             ISearchServiceClientWrapper serviceClient,
-            ISearchIndexClientWrapper searchIndexClient,
-            ISearchIndexClientWrapper hijackIndexClient,
-            IVersionListDataClient versionListDataClient,
+            Func<IBatchPusher> batchPusherFactory,
```

Comment:
>Func<IBatchPusher> batchPusherFactory [](start = 12, length = 37)

This seems to be missing from the DI container

Reply:
`Func<T>` comes for free when you register `T`.

Reply:
... which means you have to register `T`. 😆

Reply:
Whaaaaat?! I wish GitHub had a "mind blown" emoji